### PR TITLE
Fix bug where escaped back slashes were incorrectly being considered as an escaped string quote

### PR DESCRIPTION
**Pull request by sebmck**
cc @bolinfest

### CODE REVIEW

File: strip-json-comments.js [hunk 1]
```diff
@@ -19,8 +19,15 @@
 			currentChar = str[i];
 			nextChar = str[i + 1];
 
-			if (!insideComment && str[i - 1] !== '\\' && currentChar === '"') {
-				insideString = !insideString;
+			if (!insideComment && currentChar === '"') {
+				var escaped = str[i - 1] === '\\';
+				if (escaped && str[i - 2] === '\\') {
+					escaped = false;
+					insideComment = false;
```
Comment (sindresorhus) [inline strip-json-comments.js]: Why set this?

Comment (sebmck) [inline strip-json-comments.js]: Don't think I meant to leave that in.

Comment (sebmck) [inline strip-json-comments.js]: Yeah I didn't, fixed/pushed.

**Comment by sindresorhus** [comment]
Wouldn't the following be enough:

```js
var escaped = str[i - 1] === '\\' && str[i - 2] !== '\\';
if (!insideComment && !escaped && currentChar === '"') {
	insideString = !insideString;
}
```

**Comment by bolinfest** [comment]
@sindresorhus what about:

    "foo\\\""

Ideally, I don't think you should ever be considering previous characters. As you read characters in, you should make your state transitions.

**Comment by sindresorhus** [comment]
@bolinfest I assume this is what you mean:

```js
assert.strictEqual(strip('{"foo\\\"":"https://foobar.com"}'), '{"foo\\\"":"https://foobar.com"}');
```

My above suggestion passes that test though.

-

> Ideally, I don't think you should ever be considering previous characters. As you read characters in, you should make your state transitions.

I agree, but I don't have the time to refactor this at the moment. Happy to receive PRs though ;)

**Comment by sebmck** [comment]
Thanks @sindresorhus!

**Pull request merged**